### ISSUE TITLE
Snapshot plugin immediate crash with newest NS version

### ISSUE THREAD

**Issue by creambyemute** [NativeScript/android-snapshot#8]
Hi there,

We were updating our Project to angular 2.1.2 and angular/router 3.1.2 and nativescript-angular 1.1.2 and also switched to the @next version of NativeScript.

My tns info output:
┌──────────────────┬────────────────────────────┬────────────────┬─────────────┐
│ Component │ Current version │ Latest version │ Information │
│ nativescript │ 2.5.0-2016-10-31-6971 │ 2.3.0 │ Up to date │
│ tns-core-modules │ 2.5.0-2016-10-31-4629 │ 2.3.0 │ Up to date │
│ tns-android │ 2.5.0-next-2016-10-31-1332 │ 2.3.0 │ Up to date │
│ tns-ios │ 2.5.0-2016-10-31-1 │ 2.3.0 │ Up to date │
└──────────────────┴────────────────────────────┴────────────────┴─────────────┘

The problem is easily **reproducible** by doing the following:

1. tns create mySnapshotSampleApp --ng
2. tns platform add android
3. Open package.json and specify tns-android version as "next" and add tns-core-modules dependency with version "next"
4. tns install android-snapshot
5. Try to get the project running with android-snapshot, it'll always crash immediately

**Additional info**: This behaviour is reproducible in the following combination (I just tested it on a sample project):

- NativeScript@next + @angular 2.1.2 and nativescript-angular 1.1.2
- NativeScript@next + @angular 2.0.0 and nativescript-angular 1.0.1
- NativeScript 2.3 + @angular 2.1.2 and nativescript-angular 1.1.2

My package.json for the sample project:
{
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.MySampleApp",
    "tns-ios": {
      "version": "2.5.0-2016-10-31-1"
    },
    "tns-android": {
      "version": "2.5.0-next-2016-10-31-1332"
    }
  },
  "dependencies": {
    "@angular/common": "2.1.2",
    "@angular/compiler": "2.1.2",
    "@angular/core": "2.1.2",
    "@angular/forms": "2.1.2",
    "@angular/http": "2.1.2",
    "@angular/platform-browser": "2.1.2",
    "@angular/platform-browser-dynamic": "2.1.2",
    "@angular/router": "3.1.2",
    "nativescript-angular": "1.1.2",
    "nativescript-theme-core": "^0.1.3",
    "reflect-metadata": "~0.1.8",
    "rxjs": "5.0.0-beta.12",
    "tns-core-modules": "next"
  },
  "devDependencies": {
    "babel-traverse": "6.18.0",
    "babel-types": "6.18.0",
    "babylon": "6.13.1",
    "lazy": "1.0.11",
    "nativescript-dev-android-snapshot": "0.0.2",
    "nativescript-dev-typescript": "^0.3.2",
    "nativescript-dev-webpack": "0.0.20",
    "typescript": "^2.0.6",
    "zone.js": "~0.6.21"
  }
}

**Comment by jasssonpet** [comment]
Hey, @creambyemute,
We decided that managing snapshot versions of the `@next` builds of the tns-core-modules and nativescript-angular is too much of a hassle and we provide snapshots for the latest official releases only. You can try using the 2.3 version of the tns-android runtime with the 1.1.2-4.7.80 release of the nativescript-angular plugin.

**Comment by creambyemute** [comment]
@jasssonpet As I said, this also happens and is reproducible with NativeScript 2.3 (which includes tns-ios and tns-android 2.3) + @angular 2.1.2 and nativescript-angular 1.1.2 on a freshly created NS + NG2 sample app

**Comment by jasssonpet** [comment]
Ah, sorry, I missed that part. It turns out that `nativescript-angular-snapshot@1.1.2-4.7.80` was built with the wrong version of the `tns-core-modules` package. Can you try cleaning your `node_modules`/`platforms` folder, running `tns plugin add tns-core-modules@2.5.0-2016-10-24-4525` and try building again with snapshot.

**Comment by creambyemute** [comment]
@jasssonpet Okay, I'll try that. But in general, the plugin/snapshot will work on NS 2.4 after it's released I guess?

**Comment by jasssonpet** [comment]
In 2.4 we are going to make it the default option when building for Android in release config so it has to work.

**Comment by creambyemute** [comment]
Oh nice that sounds pretty good 👍, but it won't work together with webpack as described on the android performance blogpost yet I guess?

"Currently the plugin ships precompiled versions with all the Angular and Core Modules code. Instead, we can first use WebPack on the client machine, to skip all unnecessary code and then run the snapshot tool over the bundled result. This way we will be able to ship only small portion of today’s binary."

**Comment by valentinstoychev** [comment]
not yet, but the snapshot and webpack will work for 2.5 together.

**Comment by creambyemute** [comment]
Great, thanks for the clarification. Looking forward to it!

I'll try out the suggestion from @jasssonpet today and report back

**Comment by creambyemute** [comment]
Thanks, the snapshot is working when setting `tns-core-modules` to `2.5.0-2016-10-24-4525`. 

So I tried it in the real project now and it throws an exception on the login page for the `image` tag (it's reproducible when adding a login page with an image tag and an imgsrc to the sampleApp and make the default route point to that one):

`Error: Uncaught (in promise): Error: java.lang.Exception: Failed resolving constructor for class 'org.nativescript.widgets.image.Fetcher' with 1 parameters. Check the number and type of arguments.
Primitive types need to be manually wrapped in their respective Object wrappers.
If you are creating an instance of an inner class, make sure to always provide reference to the outer "this" as the first argument.`

I guess this will work when 2.4 is released, so if it's not a bug feel free to close this issue.

Edit: I just found out that this also happens without snapshot but tns-core-modules@2.5.0-2016-10-24-4525 

Thanks for your clarification and help!
